File: Core/src/main/java/dev/lynith/core/ui/theme/base/ThemeBase.java
package dev.lynith.core.ui.theme.base;

import dev.lynith.core.ui.components.impl.Button;
import dev.lynith.core.ui.theme.AbstractTheme;

public class ThemeBase extends AbstractTheme {

    public ThemeBase() {
        super("Default", ThemeType.LIGHT);
    }

    @Override
    public ButtonStyles getButtonStyles(Button button) {
        return new ButtonStyles(button);
    }

}

File: Core/src/main/kotlin/dev/lynith/core/ui/theme/base/ThemeBase.kt
package dev.lynith.core.ui.theme.base

import dev.lynith.core.ui.components.Component
import dev.lynith.core.ui.components.impl.Button
import dev.lynith.core.ui.styles.ComponentStyles
import dev.lynith.core.ui.theme.AbstractTheme

class ThemeBase : AbstractTheme(
    name = "Default",
    type = ThemeType.LIGHT,
) {

    override val colorScheme = DefaultColorScheme()

    // Custom styles for components
    override var componentMap: Map<Class<out Component<*, *>>, ComponentStyles<*, *>> = mapOf(
        Button::class.java to ButtonStyles(this),
    )


}
